### PR TITLE
HBASE-28102 [hbase-thirdparty] Bump hbase.stable.version to 2.4.17 in hbase-noop-h…

### PR DESCRIPTION
**Pull request by NihalJain**
…trace
- Also synced slf4j and junit version as per hbase 2.4.17 release
- Added mockito-all as otherwise we get `java.lang.NoClassDefFoundError: org/mockito/stubbing/Answer` at `org.apache.hadoop.hdfs.MiniDFSCluster.isNameNodeUp()` for hbase versions built with hadoop >= 3.2.4

### CODE REVIEW

**Comment by NihalJain** [comment]
Hi @Apache9 I am not very sure around the semantics of when the `hbase.stable.version` should be bumped for `hbase-noop-htrace` module. I have raised this assuming the bumping was missed some how for some time now. Please feel free to close if this is not needed.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 17s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 17s |  root in master failed.  |
| -1 :x: |  compile  |   0m 17s |  hbase-noop-htrace in master failed.  |
| -1 :x: |  javadoc  |   0m 17s |  hbase-noop-htrace in master failed.  |
||| _ Patch Compile Tests _ |
| -1 :x: |  mvninstall  |   0m 17s |  hbase-noop-htrace in the patch failed.  |
| -1 :x: |  compile  |   0m 18s |  hbase-noop-htrace in the patch failed.  |
| -1 :x: |  javac  |   0m 18s |  hbase-noop-htrace in the patch failed.  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| -1 :x: |  javadoc  |   0m 17s |  hbase-noop-htrace in the patch failed.  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 17s |  hbase-noop-htrace in the patch failed.  |
| +0 :ok: |  asflicense  |   0m 17s |  ASF License check generated no output?  |
|  |   |   3m 57s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-thirdparty/pull/104 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux d21ce2c2ed27 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| git revision | master / f82e745 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| mvninstall | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/branch-mvninstall-root.txt |
| compile | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/branch-compile-hbase-noop-htrace.txt |
| javadoc | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/branch-javadoc-hbase-noop-htrace.txt |
| mvninstall | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/patch-mvninstall-hbase-noop-htrace.txt |
| compile | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/patch-compile-hbase-noop-htrace.txt |
| javac | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/patch-compile-hbase-noop-htrace.txt |
| javadoc | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/patch-javadoc-hbase-noop-htrace.txt |
| unit | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/artifact/yetus-precommit-check/output/patch-unit-hbase-noop-htrace.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/testReport/ |
| Max. process+thread count | 9 (vs. ulimit of 1000) |
| modules | C: hbase-noop-htrace U: hbase-noop-htrace |
| Console output | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/1/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 27s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 50s |  master passed  |
| +1 :green_heart: |  compile  |   0m  6s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  8s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m  9s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  0s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  5s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |   0m 14s |  hbase-noop-htrace in the patch failed.  |
| +1 :green_heart: |  asflicense  |   0m  6s |  The patch does not generate ASF License warnings.  |
|  |   |   2m 18s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hbase.htrace_noop.TestHBaseTestingUtilityWithHTraceNoop |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/2/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-thirdparty/pull/104 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux fb130498d748 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| git revision | master / f82e745 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
| unit | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/2/artifact/yetus-precommit-check/output/patch-unit-hbase-noop-htrace.txt |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/2/testReport/ |
| Max. process+thread count | 102 (vs. ulimit of 1000) |
| modules | C: hbase-noop-htrace U: hbase-noop-htrace |
| Console output | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/2/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 25s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 51s |  master passed  |
| +1 :green_heart: |  compile  |   0m  6s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  7s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m  9s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 32s |  hbase-noop-htrace in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  6s |  The patch does not generate ASF License warnings.  |
|  |   |   2m 34s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/3/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-thirdparty/pull/104 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 477ad19f6225 5.4.0-1103-aws #111~18.04.1-Ubuntu SMP Tue May 23 20:04:10 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| git revision | master / f82e745 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/3/testReport/ |
| Max. process+thread count | 383 (vs. ulimit of 1000) |
| modules | C: hbase-noop-htrace U: hbase-noop-htrace |
| Console output | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/3/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 33s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 52s |  master passed  |
| +1 :green_heart: |  compile  |   0m  6s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  6s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 11s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 33s |  hbase-noop-htrace in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  5s |  The patch does not generate ASF License warnings.  |
|  |   |   3m 47s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/4/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-thirdparty/pull/104 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 2b1683fa9a2c 5.4.0-152-generic #169-Ubuntu SMP Tue Jun 6 22:23:09 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| git revision | master / f82e745 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/4/testReport/ |
| Max. process+thread count | 384 (vs. ulimit of 1000) |
| modules | C: hbase-noop-htrace U: hbase-noop-htrace |
| Console output | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/4/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache9** [comment]
Better to update it to the newest 2.4.x, as this version of thirdparty still needs to support hbase 2.4.x.

I think the next version will only need to support 2.5.x.

Thanks.

File: hbase-noop-htrace/pom.xml [hunk 1]
```diff
@@ -30,26 +30,29 @@
...
       -->
-    <hbase.stable.version>2.4.8</hbase.stable.version>
+    <hbase.stable.version>2.4.17</hbase.stable.version>
     <!-- should match the slf4j-api version used in the above hbase -->
-    <slf4j.version>1.7.30</slf4j.version>
+    <slf4j.version>1.7.33</slf4j.version>
     <!-- These maybe need to match the main repo -->
     <maven.checkstyle.version>3.1.0</maven.checkstyle.version>
     <checkstyle.version>8.29</checkstyle.version>
+    <!-- Added mockito-all as otherwise we get java.lang.NoClassDefFoundError:
```
Comment (NihalJain) [inline hbase-noop-htrace/pom.xml:47]: Hi @Apache9 I have retained this change, although this should not be needed when compiling with hbase 2.4.x but would be use if compiling with 2.5.x. Let me know if you want to me to drop this.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m  6s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  5s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 12s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  0s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  5s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 26s |  hbase-noop-htrace in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  4s |  The patch does not generate ASF License warnings.  |
|  |   |   2m 17s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/5/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-thirdparty/pull/104 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 891e430bc748 5.4.0-152-generic #169-Ubuntu SMP Tue Jun 6 22:23:09 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| git revision | master / f82e745 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/5/testReport/ |
| Max. process+thread count | 386 (vs. ulimit of 1000) |
| modules | C: hbase-noop-htrace U: hbase-noop-htrace |
| Console output | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/5/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 29s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m 33s |  master passed  |
| +1 :green_heart: |  compile  |   0m  6s |  master passed  |
| +1 :green_heart: |  javadoc  |   0m  5s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   0m  6s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m  5s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m  1s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  javadoc  |   0m  4s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   0m 27s |  hbase-noop-htrace in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m  4s |  The patch does not generate ASF License warnings.  |
|  |   |   2m  9s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.43 ServerAPI=1.43 base: https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/6/artifact/yetus-precommit-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-thirdparty/pull/104 |
| Optional Tests | dupname asflicense javac javadoc unit xml compile |
| uname | Linux 3c4b14351f27 5.4.0-152-generic #169-Ubuntu SMP Tue Jun 6 22:23:09 UTC 2023 x86_64 GNU/Linux |
| Build tool | maven |
| git revision | master / f82e745 |
| Default Java | Oracle Corporation-1.8.0_282-b08 |
|  Test Results | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/6/testReport/ |
| Max. process+thread count | 405 (vs. ulimit of 1000) |
| modules | C: hbase-noop-htrace U: hbase-noop-htrace |
| Console output | https://ci-hbase.apache.org/job/HBase-Thirdparty-PreCommit/job/PR-104/6/console |
| versions | git=2.20.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by Apache9 — APPROVED** [review]

**Pull request merged**